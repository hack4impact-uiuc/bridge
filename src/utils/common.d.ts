declare module "@hack4impact-uiuc/bridge/common" {
  import { Props, ReactType } from "react";
  import {
    ColorProps,
    SpaceProps,
    LayoutProps,
    TypographyProps as StyledTypographyProps,
    BordersProps as StyledBordersProps,
    BoxShadowProps,
  } from "styled-system";

  export interface BaseProps extends Props<any> {
    as?: ReactType;
    className?: string;
  }
  export interface CommonProps
    extends BaseProps,
      ColorProps,
      SpaceProps,
      LayoutProps {}

  export interface TypographyProps extends BaseProps, StyledTypographyProps {}

  export interface BorderProps extends StyledBordersProps, BoxShadowProps {}
}
