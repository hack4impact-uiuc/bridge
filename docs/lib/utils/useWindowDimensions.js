import { useState, useEffect } from 'react';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useWindowDimensions() {
  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());

  useEffect(() => {
    function handleResize() {
      const { innerWidth: width, innerHeight: height } = window;
      setWindowDimensions({ width, height });
    }

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowDimensions;
}
