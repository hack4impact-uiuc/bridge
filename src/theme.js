/* eslint-disable prefer-destructuring */

function fontStack(fonts) {
  return fonts.map((font) => (font.includes(' ') ? `"${font}"` : font)).join(', ');
}

const DEFAULT_SHADOW = '0 1px 6px 2px rgba(0,0,0,0.15)';

const baseColors = {
  hackBlue: ['#B4D8FA', '#3E87CD', '#155DA1', '#0E4E8A', '#0D3F6E', '#031425'],
  indigoBlue: ['#BCD0F3', '#415F94', '#2D4979', '#253E68', '#13294E', '#061734'],
  impactGrey: ['#FAFBFC', '#EBEEF2', '#8B9199', '#666B72', '#46494F'],
};

const hackBluePalette = {
  lighter: baseColors.hackBlue[0],
  light: baseColors.hackBlue[1],
  primary: baseColors.hackBlue[2],
  dark: baseColors.hackBlue[3],
  darker: baseColors.hackBlue[4],
  text: baseColors.hackBlue[5],
};

const indigoPalette = {
  lighter: baseColors.indigoBlue[0],
  light: baseColors.indigoBlue[1],
  primary: baseColors.indigoBlue[2],
  dark: baseColors.indigoBlue[3],
  darker: baseColors.indigoBlue[4],
  text: baseColors.indigoBlue[5],
};

const impactGreyPalette = {
  lighter: baseColors.impactGrey[0],
  light: baseColors.impactGrey[1],
  midtone: baseColors.impactGrey[2],
  primary: baseColors.impactGrey[3],
  darker: baseColors.impactGrey[4],
};

const secondaryColors = {
  green: ['#BDF3C6', '#90E19E', '#4C9859', '#256830', '#154C1E', '#031D07'],
  red: ['#FFBCBF', '#D1595F', '#A62B31', '#8A1A1F', '#6B1014', '#35080A'],
  yellow: ['#FFECBD', '#FFD15F', '#FDBE21', '#DA9C03', '#A77700', '#201803'],
};

const greenPalette = {
  lighter: secondaryColors.green[0],
  light: secondaryColors.green[1],
  primary: secondaryColors.green[2],
  dark: secondaryColors.green[3],
  darker: secondaryColors.green[4],
  text: secondaryColors.green[5],
};

const redPalette = {
  lighter: secondaryColors.red[0],
  light: secondaryColors.red[1],
  primary: secondaryColors.red[2],
  dark: secondaryColors.red[3],
  darker: secondaryColors.red[4],
  text: secondaryColors.red[5],
};

const yellowPalette = {
  lighter: secondaryColors.yellow[0],
  light: secondaryColors.yellow[1],
  primary: secondaryColors.yellow[2],
  dark: secondaryColors.yellow[3],
  darker: secondaryColors.yellow[4],
  text: secondaryColors.yellow[5],
};

const typographyColors = {
  charcoal: '#272A2D',
  iron: '#4B4E52',
  ink: '#0A162A',
  slate: '#363949',
};

const colors = {
  black: '#000000',
  white: '#FFFFFF',
  ...baseColors,
  ...secondaryColors,
  ...typographyColors,
  hackBluePalette,
  indigoPalette,
  impactGreyPalette,
  greenPalette,
  redPalette,
  yellowPalette,
  text: {
    grayScale: {
      main: typographyColors.charcoal,
      support: typographyColors.iron,
    },
    coolTone: {
      main: typographyColors.ink,
      support: typographyColors.slate,
    },
  },
  // states or in other words color variants
  variants: {
    primary: hackBluePalette,
    success: greenPalette,
    error: redPalette,
    warning: yellowPalette,
    secondary: hackBluePalette,
    white: {
      primary: '#FFFFFF',
      dark: '#FFFFFF',
      lighter: indigoPalette.light,
    },
  },
};

const fontSizes = ['12px', '14px', '16px', '18px', '20px', '22px', '25px', '28px', '32px', '45px', '60px'];

const fonts = {
  chivo: fontStack(['Chivo']),
  hkGrotesk: fontStack(['HKGrotesk']),
};

const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  bold: 700,
};

// styling specific to the Button component
const button = {
  font: fonts.chivo,
  fontWeight: fontWeights.bold,
  fontSize: '14px',
  letterSpacing: '2px',
  // default values, based on the color palette
  default: (colorPalette) => ({
    bg: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
      disabled: colorPalette.lighter,
    },
    border: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colorPalette.light,
    },
    shadow: {
      default: 'none',
      hover: 'none',
      disabled: 'none',
    },
    outline: {
      bg: {
        default: colors.white,
        hover: colors.white,
        disabled: colors.white,
      },
      border: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      fontColor: {
        default: colorPalette.primary,
        hover: colorPalette.dark,
        disabled: colorPalette.lighter,
      },
      shadow: {
        default: 'none',
        hover: 'none',
        disabled: 'none',
      },
    },
  }),

  // everything below overrides the default values
  primary: {
    fontColor: {
      default: colors.white,
      hover: colors.white,
      disabled: colors.white,
    },
  },
  white: {
    fontColor: {
      default: indigoPalette.primary,
      hover: indigoPalette.primary,
      disabled: indigoPalette.lighter,
    },
    shadow: {
      hover: DEFAULT_SHADOW,
    },
    outline: {
      bg: {
        default: indigoPalette.primary,
        hover: indigoPalette.primary,
        disabled: indigoPalette.primary,
      },
      border: {
        disabled: indigoPalette.light,
      },
      fontColor: {
        default: colors.white,
        hover: colors.white,
        disabled: indigoPalette.lighter,
      },
      shadow: {
        hover: DEFAULT_SHADOW,
      },
    },
  },
  secondary: {
    bg: {
      hover: hackBluePalette.darker,
      disabled: impactGreyPalette.light,
    },
    fontColor: {
      disabled: impactGreyPalette.midtone,
    },
    outline: {
      bg: {
        disabled: colors.white,
      },
      border: {
        hover: hackBluePalette.darker,
        disabled: impactGreyPalette.light,
      },
      fontColor: {
        disabled: indigoPalette.lighter,
      },
    },
  },
};


const link = {
  default: (colorPalette) => ({
    color: {
      default: colorPalette.primary,
      hover: colorPalette.dark,
    },
  }),
  secondary: {
    default: hackBluePalette.primary,
    hover: hackBluePalette.darker,
  },
};


const theme = {
  // General
  colors,
  fonts,
  fontWeights,
  fontSizes,
  button,
  link,
  DEFAULT_SHADOW,
};


export default theme;
export { colors };
