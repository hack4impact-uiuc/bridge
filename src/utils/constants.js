import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';
import PropTypes from 'prop-types';


const { compose, system } = styledSystem;

// Text Decoration Prop
export const textDecoration = system({
  prop: 'textDecoration',
  cssProperty: 'text-decoration',
});
textDecoration.propTypes = PropTypes.string;


// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.layout,
};

// for easier access
export const TYPOGRAPHY = styledSystem.typography;
export const LAYOUT = styledSystem.layout;
export const POSITION = styledSystem.position;
export const FLEX = styledSystem.flexbox;
export const GRID = styledSystem.grid;
export const COLOR = styledSystem.color;
export const BORDER = styledSystem.border;

TYPOGRAPHY.propTypes = systemPropTypes.typography;
LAYOUT.propTypes = systemPropTypes.layout;
POSITION.propTypes = systemPropTypes.position;
FLEX.propTypes = systemPropTypes.flexbox;
GRID.propTypes = systemPropTypes.grid;
COLOR.propTypes = systemPropTypes.color;
BORDER.propTypes = systemPropTypes.border;
