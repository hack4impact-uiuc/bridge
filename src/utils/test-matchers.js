/* eslint-disable react/forbid-foreign-prop-types */
// taken from https://github.com/primer/components/blob/master/src/utils/test-matchers.js
import 'jest-styled-components';

const stringify = (d) => JSON.stringify(d, null, '  ');
/**
 * These are props that styled-system aliases for backwards compatibility.
 * For some reason, they don't show up in our toImplementSystemProps() matcher,
 * so we skip over them.
 */
const ALIAS_PROP_TYPES = ['w', 'align', 'justify', 'wrap'];

// eslint-disable-next-line no-undef
expect.extend({
  toImplementSystemProps(Component, propNames) {
    const propKeys = new Set(Object.keys(Component.propTypes));
    const expectedPropKeys = Object.keys(propNames.propTypes);
    const missing = expectedPropKeys.filter((key) => !propKeys.has(key))
      .filter((key) => !ALIAS_PROP_TYPES.includes(key));
    return {
      pass: missing.length === 0,
      message: () => `Missing prop${missing.length === 1 ? '' : 's'}: ${stringify(missing)}`,
    };
  },
});
