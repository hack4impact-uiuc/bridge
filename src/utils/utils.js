import themeGet from '@styled-system/theme-get';
import * as styledSystem from 'styled-system';
import theme from '../theme';

const { get: getKey } = styledSystem;

export const get = (key) => themeGet(key, getKey(theme, key));

export const getOrDefault = (val, def) => (val !== undefined && val !== null ? val : def);

export const lodashGet = (object, path, defaultVal) => {
  const PATH = Array.isArray(path)
    ? path
    : path.split('.').filter((i) => i.length);
  if (!PATH.length) {
    return object === undefined ? defaultVal : object;
  }
  if (object === null || object === undefined || typeof (object[PATH[0]]) === 'undefined') {
    return defaultVal;
  }
  return lodashGet(object[PATH.shift()], PATH, defaultVal);
};
