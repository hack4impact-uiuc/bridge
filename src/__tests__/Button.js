import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';

import { Button } from '..';
import theme, { colors } from '../theme';
import { renderJSON } from '../utils/testing';

const noop = () => {};


describe('Button', () => {
  it('renders a <button>', () => {
    expect(renderJSON(<Button />).type).toEqual('button');
  });

  it('renders children', () => {
    const buttonText = 'APPLY';
    render(<Button>{buttonText}</Button>);
    expect(screen.getByText(buttonText)).toBeInTheDocument();
  });

  it('preserves "onClick" prop', () => {
    expect(renderJSON(<Button onClick={noop} />).props.onClick).toEqual(noop);
  });

  it('respects "width" props', () => {
    const { container } = render(<Button width={200} />);
    expect(container.firstChild).toHaveStyle('width: 200px;');
  });

  it('respects the "disabled" prop', () => {
    const item = renderJSON(<Button disabled />);
    expect(item.props.disabled).toEqual(true);
    expect(item).toMatchSnapshot();
  });

  it('respects "outline" prop', () => {
    const item = renderJSON(<Button outline />);
    expect(item).toMatchSnapshot();
  });

  it('renders primary color as default', () => {
    const { container } = render(<Button>APPLY</Button>);
    expect(container.firstChild).toHaveStyle(`color: ${theme.button.primary.fontColor.default}`);
    expect(container.firstChild).toHaveStyle(`background: ${theme.button.default(colors.hackBluePalette).bg.default}`);
  });

  it('renders correct border/bg/color with "outline" prop (primary colors)', () => {
    const { container } = render(<Button outline>APPLY</Button>);
    expect(container.firstChild).toHaveStyle(`color: ${theme.button.default(colors.hackBluePalette).outline.fontColor.default}`);
    expect(container.firstChild).toHaveStyle(`background: ${theme.button.default(colors.hackBluePalette).outline.bg.default}`);
    expect(container.firstChild).toHaveStyle(`border: 2px solid ${theme.button.default(colors.hackBluePalette).outline.border.default}`);
  });

  it('renders correct colors with "outline" and "disabled" props (primary colors)', () => {
    const { container } = render(<Button disabled outline>APPLY</Button>);
    expect(container.firstChild).toHaveStyle(`color: ${theme.button.default(colors.hackBluePalette).outline.fontColor.disabled}`);
    expect(container.firstChild).toHaveStyle(`background: ${theme.button.default(colors.hackBluePalette).outline.bg.disabled}`);
    expect(container.firstChild).toHaveStyle(`border: 2px solid ${theme.button.default(colors.hackBluePalette).outline.border.disabled}`);

    const item = renderJSON(<Button disabled outline />);
    expect(item).toMatchSnapshot();
  });
});

// Colors should match colors in default theme (design system styles)
describe('Button Colors', () => {
  it('renders primary colors', () => {
    const item = renderJSON(<Button variant="primary" />);
    expect(item).toMatchSnapshot();
  });

  it('renders secondary colors', () => {
    const item = renderJSON(<Button variant="secondary" />);
    expect(item).toMatchSnapshot();
  });

  it('renders success colors', () => {
    const item = renderJSON(<Button variant="success" />);
    expect(item).toMatchSnapshot();
  });

  it('renders error colors', () => {
    const item = renderJSON(<Button variant="error" />);
    expect(item).toMatchSnapshot();
  });

  it('renders warning colors', () => {
    const item = renderJSON(<Button variant="warning" />);
    expect(item).toMatchSnapshot();
  });
});
