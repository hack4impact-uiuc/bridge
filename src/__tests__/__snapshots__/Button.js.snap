// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Colors renders error colors 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #A62B31;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #8A1A1F;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #FFBCBF;
  border: none;
  box-shadow: none;
  color: #D1595F;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Colors renders primary colors 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0E4E8A;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #B4D8FA;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Colors renders secondary colors 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0D3F6E;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #EBEEF2;
  border: none;
  box-shadow: none;
  color: #8B9199;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Colors renders success colors 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #4C9859;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #256830;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #BDF3C6;
  border: none;
  box-shadow: none;
  color: #90E19E;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Colors renders warning colors 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #FDBE21;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #DA9C03;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #FFECBD;
  border: none;
  box-shadow: none;
  color: #FFD15F;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Sizes renders a large button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 14px 32px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0E4E8A;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #B4D8FA;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
}

<button
  className="c0"
  type="large"
/>
`;

exports[`Button Sizes renders a medium button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0E4E8A;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #B4D8FA;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button Sizes renders a small button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 12px;
  line-height: 16px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0E4E8A;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #B4D8FA;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
}

<button
  className="c0"
  type="small"
/>
`;

exports[`Button renders correct colors with "outline" and "disabled" props (primary colors) 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #FFFFFF;
  border: 2px solid #155DA1;
  color: #155DA1;
}

.c0:hover:not([disabled]) {
  background: #FFFFFF;
  border: 2px solid #0E4E8A;
  box-shadow: none;
  color: #0E4E8A;
  cursor: pointer;
}

.c0:disabled {
  background: #FFFFFF;
  border: 2px solid #B4D8FA;
  box-shadow: none;
  color: #B4D8FA;
}

<button
  className="c0"
  disabled={true}
  type="medium"
/>
`;

exports[`Button respects "outline" prop 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #FFFFFF;
  border: 2px solid #155DA1;
  color: #155DA1;
}

.c0:hover:not([disabled]) {
  background: #FFFFFF;
  border: 2px solid #0E4E8A;
  box-shadow: none;
  color: #0E4E8A;
  cursor: pointer;
}

.c0:disabled {
  background: #FFFFFF;
  border: 2px solid #B4D8FA;
  box-shadow: none;
  color: #B4D8FA;
}

<button
  className="c0"
  type="medium"
/>
`;

exports[`Button respects the "disabled" prop 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  border-radius: 13px;
  font-family: Chivo;
  font-weight: 700px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 16px;
  display: inline-block;
  text-align: center;
  vertical-align: middle;
  background: #155DA1;
  border: none;
  color: #FFFFFF;
}

.c0:hover:not([disabled]) {
  background: #0E4E8A;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
  cursor: pointer;
}

.c0:disabled {
  background: #B4D8FA;
  border: none;
  box-shadow: none;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={true}
  type="medium"
/>
`;
